jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactoryInternals.java

```diff
@@ -52,6 +52,7 @@
         PathResourceFactory pathResourceFactory = new PathResourceFactory();
         RESOURCE_FACTORIES.put("file", pathResourceFactory);
         RESOURCE_FACTORIES.put("jrt", pathResourceFactory);
+        RESOURCE_FACTORIES.put("resource", pathResourceFactory); // GraalVM native-image resource
```

Comment:
This shall not be automatic or defaulted.
The modifications you have made to `URIUtil` and `PathResource` should be refactored out into a standalone `GraalVMResourceFactory` implementation, not by modifying the default behaviors.

Reply:
I've thought about that, but then GraalVM-unaware code that simply tries to serve resources from the classpath will fail.

I've amended my change to transparently support the following real-world use case:

    URL resourceURL = MyClass.class.getResource("some/resource");
    Resource resource = ResourceFactory.root().newResource(resourceURL);

The new change only enables "resource:" URLs when a well-known GraalvM Native Image system property is set.

Reply:
We can go one of two ways:
1)  automatic discovery of unknown schemes.  We have always done this for URL handlers, but now it looks like we'd also need to do an attempt to to look for a filesystem for a Path based unknown scheme provided by the JVM.  That makes me less inclined to go automatic.
2) A simple registration mechanism, that can easily be put into a module.  Potentially fairly generic so it can add either URL handlers or FileSystem mounters for an arbitrary resource.   A graalvm module would then do the mapping for "resource" to the FileSystem mounting extension of PathResourceFactory.